polynomials.go

```diff
@@ -160,11 +161,20 @@ const randPolMaxTries = 1e6
 // "Fingerprinting by Random Polynomials", page 4. If no polynomial could be
 // found in one million tries, an error is returned.
 func RandomPolynomial() (Pol, error) {
+	return DeterministicPolynomial(rand.Reader)
+}
+
+// DeterministicPolynomial uses a source of bytes to generate a polynomial.
+// The source should appear pseudorandom, at least, and by default would be
+// rand.Reader. For a secure deterministic polynomial, it could instead
+// be the output of the SHA3/SHAKE algorithm after seeding securely with
+// secret input.
+func DeterministicPolynomial(source io.Reader) (Pol, error) {
```

Comment:
I don't like the documentation for this function: What exactly does "appear pseudorandom" mean? It's not random, not even pseudo-random, but only "appears" to be?

I'd like to suggest naming this function `DerivePolynomial(source io.Reader)`, with a description along the lines of "Returns an irreducible Polynomial generated by reading random bytes from source", followed by the documentation for the `RandomPolynomial` function.